pkg/apb/executor.go

```diff
@@ -76,20 +88,49 @@ func ExecuteApb(
 						extraVars,
 					},
 					ImagePullPolicy: pullPolicy,
+					VolumeMounts:    volumeMounts,
 				},
 			},
 			RestartPolicy:      v1.RestartPolicyNever,
-			ServiceAccountName: serviceAccountName,
+			ServiceAccountName: executionContext.ServiceAccount,
+			Volumes:            volumes,
 		},
 	}
 
-	log.Notice(fmt.Sprintf("Creating pod %q in the %s namespace", pod.Name, ns))
+	log.Notice(fmt.Sprintf("Creating pod %q in the %s namespace", pod.Name, executionContext.Namespace))
 	k8scli, err := clients.Kubernetes(log)
 	if err != nil {
-		return apbID, err
+		return executionContext, err
+	}
+	_, err = k8scli.CoreV1().Pods(executionContext.Namespace).Create(pod)
+
+	return executionContext, err
```

Comment:
Yeah that was my instinct as well, the only reason I'm returning the partial execution context is because the function was returning some values and an error already, and I didn't want to break anything that may have been using it.